### PR TITLE
HMAN-80 Establish and validate PUT /hearingActuals

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-80

### Change description ###
Established PUT /hearingActuals endpoint with validation.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        validateHearingActualDaysNotInTheFuture(request);
+        validateDuplicateHearingActualDays(request);
+        validateHearingActualDaysNotBeforeFirstHearingDate(request, hearing);
+        validateHearingResult(request);
+    }
+
+    private void validateHearingResult(HearingActual request) {
+        if (HEARING_RESULTS_THAT_NEED_REASON_TYPE.contains(request.getHearingOutcome().getHearingResult().toUpperCase())
+            && StringUtils.isBlank(request.getHearingOutcome().getHearingResultReasonType())) {
+            throw new BadRequestException(null);
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java:60]: why null message

Comment (rafalkalita) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java:60]: There was no message defined in the AC. But after a chat with the BA we have decided to put one in here.

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    private void validateHearingStatusForActuals(String hearingStatus) {
+        if (ALLOWED_ACTUALS_STATUSES.stream().noneMatch(e -> e.equals(hearingStatus))) {
+            throw new BadRequestException(String.format(HEARING_ACTUALS_INVALID_STATUS, hearingStatus));
+        }
+    }
+
+    private HearingEntity getHearing(Long hearingId) {
+        Optional<HearingEntity> hearingEntityOptional = hearingRepository.findById(hearingId);
+
+        if (hearingEntityOptional.isEmpty()) {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java:102]: this validation exists in multiple places, can we move this to common place

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingHelper.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.groupingBy;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.INVALID_HEARING_STATE;
+
+@Component
+public class HearingHelper extends HearingIdValidator {
```
Comment (kiran-yenigala-hmcts) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/HearingHelper.java:21]: can we use composition of HearingIdValidator instead of extending it

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Minor Comments

**Pull request merged**